src/analysis/normed_space/inner_product.lean

```diff
@@ -2216,3 +2316,75 @@ begin
   exact nat.add_sub_cancel' (nat.succ_le_iff.mpr findim_pos)
 end
 end orthogonal
+
+section orthonormal_basis
+
+/-! ### Existence of an orthonormal basis for a finite-dimensional inner product space -/
+
+variables (𝕜 E)
+
+open finite_dimensional
+
+/-- A finite-dimensional `inner_product_space` has an orthonormal set whose cardinality is the
```

Comment:
There is the more general statement that, in any inner product space, there is an orthonormal set spanning a dense subspace of the space (proof: take a maximal such family, by Zorn. If its span where not dense, take a nonzero vector in the orthogonal and add it to the family, giving a contradiction) (this can be written as `orthonormal 𝕜 (λ (x : s), (x : E))` so no need to introduce a new formalism). This is important because it gives Hilbert bases in Hilbert spaces. And the statement below is essentially a specialization of this -- although there is some bureaucracy to go from one to the other. Do you think it is worth adding it?

Reply:
I'll try!  It'll be fun to try a Zorn proof.